lib/Basics/FileUtils.cpp

```diff
@@ -739,80 +740,42 @@ void makePathAbsolute(std::string& path) {
   }
 }
 
-static void throwProgramError(std::string const& filename) {
-  auto res = TRI_set_errno(TRI_ERROR_SYS_ERROR);
-
-  LOG_TOPIC("a557b", TRACE, arangodb::Logger::FIXME)
-      << StringUtils::concatT("open failed for file '", filename, "': ", TRI_last_error());
-
-  THROW_ARANGO_EXCEPTION(res);
-}
-
 std::string slurpProgram(std::string const& program) {
-#ifdef _WIN32
-  icu::UnicodeString uprog(program.c_str(), static_cast<int32_t>(program.length()));
-  FILE* fp = _wpopen(reinterpret_cast<wchar_t const*>(uprog.getTerminatedBuffer()), L"r");
-#else
-  FILE* fp = popen(program.c_str(), "r");
-#endif
-
-  constexpr size_t chunkSize = 8192;
-  StringBuffer buffer(chunkSize, false);
-
-  if (fp) {
-    int c;
-
-    while ((c = getc(fp)) != EOF) {
-      buffer.appendChar((char)c);
-    }
-
-#ifdef _WIN32
-    int res = _pclose(fp);
-#else
-    int res = pclose(fp);
-#endif
+  ExternalProcess const* process;
+  ExternalId external;
+  ExternalProcessStatus res;
+  std::string output;
+  std::vector<std::string> moreArgs;
+  std::vector<std::string> additionalEnv;
+  char buf[1024];
+
+  moreArgs.push_back(std::string("version"));
+
+  TRI_CreateExternalProcess(program.c_str(),
+                            moreArgs,
+                            additionalEnv,
+                            true,
+                            &external);
+  if (external._pid == TRI_INVALID_PROCESS_ID) {
+    auto res = TRI_set_errno(TRI_ERROR_SYS_ERROR);
 
-    if (res != 0) {
-      throwProgramError(program);
+    LOG_TOPIC("a557b", TRACE, arangodb::Logger::FIXME)
+      << StringUtils::concatT("open failed for file '",
+                              program, "': ",
+                              TRI_last_error());
+    THROW_ARANGO_EXCEPTION(res);
+  }
+  process = TRI_LookupSpawnedProcess(external._pid);
+  while (res = TRI_CheckExternalProcess(external, false, 0),
+         (res._status == TRI_EXT_RUNNING)) {
+    auto nRead = TRI_ReadPipe(process, buf, sizeof(buf) - 1);
+    if (nRead > 0) {
+      output += std::string(buf, nRead);
```

Comment:
Can be made more efficient without creating a temporary string:
```suggestion
      output.append(buf, nRead);
```